lte/gateway/c/session_manager/SessionState.cpp

```diff
@@ -221,12 +225,22 @@ void SessionState::insert_pdr(SetGroupPDR* rule) {
   PdrList_.push_back(*rule);
 }
 
-void SessionState::set_remove_all_pdrs() {
+/* method to change the PDR state */
+void SessionState::set_all_pdrs(enum PdrState pdstate) {
   for (auto& rule : PdrList_) {
-    rule.set_pdr_state(PdrState::REMOVE);
+    rule.set_pdr_state(pdstate);
   }
 }
-/* Remove all Pdr, FAR rules */
+
+/* method to search specific pdr id existense */
+bool SessionState::search_pdr(unsigned int id) {
```

Comment:
How often is this lookup performed? Should we change the underlying datastructure to a map of ID -> PDRList?

Reply:
Good alternative approach to think. As elements are going to small, I used a vector here, however we cud use maps, not sure the entries are going to be more than 5 entries for a session. For now I am leaving it as it is.

Reply:
but good alternative approach think based on the need?

Reply:
Sounds good then, it might be good to leave it as a comment in the code.